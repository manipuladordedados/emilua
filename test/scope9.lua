-- Coroutine scopes are fiber independent

local println = require('println')

co = coroutine.create(function()
    scope_cleanup_push(function() println('c1') end)
    scope_cleanup_push(function() println('c2') end)
    scope_cleanup_push(function() println('c3') end)
    coroutine.yield()
    scope_cleanup_pop()
    coroutine.yield()
    scope_cleanup_pop()
    coroutine.yield()
    scope_cleanup_pop()
end)

spawn(function()
    scope_cleanup_push(function() println('p1') end)
    coroutine.resume(co)
end):detach()

spawn(function()
    scope_cleanup_push(function() println('p2') end)
    coroutine.resume(co)
end):detach()

spawn(function()
    scope_cleanup_push(function() println('p3') end)
    coroutine.resume(co)
end):detach()

coroutine.resume(co)
