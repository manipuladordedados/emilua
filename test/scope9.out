c3
p1
c2
p2
c1
p3
